cudamat/cudamat.py

```diff
@@ -1501,6 +1503,25 @@ def where(condition_mat, if_mat, else_mat, target=None):
     return target
 
 
+def filter(mat, filter, target=None):
+    """
+    Apply a square filter matrix to the matrix mat.
```

Comment:
square -> rectangular (or without any adjective), since you don't actually require the filter to be square